components/ide/theia/leeway.Dockerfile

```diff
@@ -24,47 +23,61 @@ RUN apk add --no-cache bash gcc g++ make pkgconfig python libc6-compat libexecin
 COPY --from=node_installer /theia/node/ /theia/node/
 ENV PATH=$PATH:/theia/node/bin/
 
+
 # install yarn by download+unpack to ensure it does NOT put anything into /theia/node/
 RUN wget https://github.com/yarnpkg/yarn/releases/download/v1.15.2/yarn-v1.15.2.tar.gz
 RUN tar zvxf yarn-v1.15.2.tar.gz
 ENV PATH=$PATH:/yarn-v1.15.2/bin/
 
-COPY components-theia-app--installer /theia-installer
+
+
+
 WORKDIR /theia
-RUN /theia-installer/install.sh
+
+ENV THEIA_APP_COMMIT c3571dc9745812a20798c9542bb5e4fdfd4804fc
+RUN mkdir theia-app \
+    && cd theia-app \
+    && git init \
+    && git remote add origin https://github.com/gitpod-io/theia-app \
+    && git fetch origin $THEIA_APP_COMMIT \
+    && git reset --hard FETCH_HEAD
+
+
+WORKDIR /theia/theia-app
+RUN yarn
+RUN yarn build
+
 
 # copy native dependencies of node
 COPY package-libs.sh /usr/bin/
 RUN package-libs.sh /theia/node/bin/node
 
-# patching plugin ext host process to fix css extension compatibility, see https://github.com/gitpod-io/gitpod/pull/2483
-RUN cp -r /theia/node_modules/@gitpod/gitpod-ide/patches /theia/patches 
-RUN yarn patch-package
-
 # copy native dependencies of node modules
-RUN find /theia/node_modules/ -iname *.node -exec package-libs.sh {} \;
+RUN find /theia/theia-app/node_modules/ -iname *.node -exec package-libs.sh {} \;
 
 RUN cp /theia/node/bin/node /theia/node/bin/gitpod-node && rm /theia/node/bin/node
 
+# patching plugin ext host process to fix css extension compatibility, see https://github.com/gitpod-io/gitpod/pull/2483
+# RUN cp -r /theia/theia-app/node_modules/@gitpod/gitpod-ide/patches /theia/theia-app/patches 
```

Comment:
TBD

Reply:
It should be moved and kept.

Reply:
fully agree, see the todo from ☝🏻 
just, ditching leeway build led to different directory structures, I then took a shortcut and left a todo.